### PR TITLE
added the checking of insert into provider's stack

### PR DESCRIPTION
**Pull request by avoget**
Code to fix collect_providers method to return the result of sk_OSSL_PROVIDER_push.
The result is using in OSSL_PROVIDER_do_all and protects it from memory limitation error.

### CODE REVIEW

File: apps/list.c [hunk 1]
```diff
@@ -1209,9 +1209,11 @@ static int provider_cmp(const OSSL_PROVIDER * const *a,
 static int collect_providers(OSSL_PROVIDER *provider, void *stack)
 {
     STACK_OF(OSSL_PROVIDER) *provider_stack = stack;
-
-    sk_OSSL_PROVIDER_push(provider_stack, provider);
-    return 1;
+    /*
+	 * If OK - result is the index of inserted data
```
Comment (paulidale) [inline apps/list.c:1213]: Please, no tabs in files they are evil as the misindentation I'm seeing indicates :)

**Review by paulidale — COMMENTED** [review]
Apart from the tabs, this looks good.

File: apps/list.c (hunk 1 above)
Comment (avoget) [reply apps/list.c]: Hello Pauli.
Tabs are removed.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
**Note for committers when merging: please squash the commits.**

**Comment by avoget** [comment]
@t8m 
Hello Tomas.

What should I do to merge this PR ?

Regards,
Alexey

**Comment by mattcaswell** [comment]
This will be merged by a committer in due course.

**Comment by avoget** [comment]
Hello Matt.
Thank you !

**Comment by t8m** [comment]
Squashed and merged to the master, 3.2, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**